lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -24,12 +24,26 @@ function snakeCaseToCamelCase(str) {
 }
 
 class FullPageScreenshot extends Gatherer {
+  /**
+   * @param {Driver} driver
+   * @return {Promise<number>}
+   * @see https://bugs.chromium.org/p/chromium/issues/detail?id=770769
+   */
+  async getMaxScreenshotHeight(driver) {
+    return await driver.evaluate(pageFunctions.getMaxTextureSize, {
```

Comment:
curious why a page function and not local to this module?

Reply:
mostly because theres a lot of ways in which PFs are very diff than node code and so i'd like to ideally have all PFs live in separate files.  

* code coverage exemption (istanbul ignore/c8 ignore) is easier since they're colocated
* can apply a `eslint-env browser` instead of our current handcrafted `/* global window document Node ShadowRoot */` list.
* maybe use some browserish config for typescript on these files

--

but until all are migrated there i'll just place new ones in pf.js. thats my thinking.

i don't care a lot about this particular case tho.

Reply:
> but until all are migrated there i'll just place new ones in pf.js. thats my thinking.

Funny, I have the exact opposite thinking. Not speaking to the idea at all, but if we are currently doing "all page functions local if possible", we should keep doing that until we change how we structure things, instead of starting a transitionary thing w/o any discussion.

Reply:
word.

well... looking at the fn's in pf.js, there already are plenty that are one-offs used by gatherrunner & driver. oh and a singleuse for iframe-elements!  :)